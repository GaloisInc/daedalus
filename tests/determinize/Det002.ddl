-- testing LL(1) with Match and some semantic value

def Main = { x = Simple1 ; y = Simple2 }

def Simple1 =
    {@x = Match "ab"; ^ concat [ x, x ]}
  | Match "bcde"

def Simple2 =
    Match "abc"
  --| Match "ac"
  | Match "bc"
  | { @x = Match1 'c' ; @y = Match "b"; ^ (concat [ [x], y]) }


