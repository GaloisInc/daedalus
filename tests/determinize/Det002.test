dump
Det002.ddl
--core
--determinize
