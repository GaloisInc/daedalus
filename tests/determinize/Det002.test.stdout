module DaedalusMain where
 
 
type Main =
  struct
    x : [Word 8]
    y : [Word 8]
 
-------------
 
proc Simple1() : [Word 8] =
  -- DETERMINIZE HERE
  do _x0 <- match1 { ... }
     case _x0 of
       97 ->
         do x <- do match_ "b"
                    pure "ab"
            pure (concat [x, x])
       98 ->
         do match_ "cde"
            pure "bcde"
       _ ->
         fail_sys @[Word 8]
 
proc Simple2() : [Word 8] =
  -- DETERMINIZE HERE
  do _x0 <- match1 { ... }
     case _x0 of
       97 ->
         do match_ "bc"
            pure "abc"
       98 ->
         do match_ "c"
            pure "bc"
       99 ->
         do x <- pure (99 @(Word 8))
            y <- match "b"
            pure (concat [[x], y])
       _ ->
         fail_sys @[Word 8]
 
proc Main() : Main =
  do x <- Simple1()
     y <- Simple2()
     pure Main {x = x, y = y}
